R/variance.R

```diff
@@ -10,3 +12,194 @@ vcovHC.prediction_cf <- function(x, type = "HC3",...) {
   dimnames(ret) <- list(names(x), names(x))
   ret
 }
+
+#' Covariance matrix of Counterfactual prediction
+#'
+#' @param prediction_cf R object returned by \code{\line{predict_counterfactual}}
+#' @param eff_measure default is NULL, the covariance matrix of the mean counterfactual
+#' predictions of the response. If not NULL, a valid option is "diff", "odds ratio", or "risk ratio",
+#' in which case covariance matrix of the estimated effect measure will also be returned.
+#'
+#' @return an R object of class 'vcov_robincar', essentially a list consisting
+#' \itemize{
+#'   \item theta: mean counterfactual prediction of the reponse, as the calculation result of
+#'         \code{\line{predict_counterfactual}}
+#'   \item V: covariance matrix of the mean counterfactual predictions of the response
+#'   \item f_vcov: covariance matrix of the estimated effect measures. It is NULL
+#'         if \code{eff_measure} is set to NULL (default)
+#'   \item n: sample size in \code{fit} of \code{\line{predict_counterfactual}}
+#'   \item eff_measure: \code{eff_measure} in the argument of this function
+#'   \item fit: \code{fit} of \code{\line{predict_counterfactual}}
+#'   \item trt_var_name: name of treatment variable in the data used in \code{fit}
+#'         of \code{\line{predict_counterfactual}}
+#' }
+#' @export
+
+vcovRobinCar.prediction_cf <- function(x, eff_measure = NULL, ...) {
```

Comment:
lets' avoid the dots in the function names (this is not a method; the name `vcovHC.prediction_cf` is fine because `vcovHC` is a generic in `sandwich`

Reply:
or alternatively, we can also make `vcovHC.prediction_cf` another name